crates/core/src/interaction.rs

```diff
@@ -0,0 +1,12 @@
+#[derive(Debug)]
+pub enum InteractionType {
```

Comment:
Could we derive `Serialize` and `Deserialize` for these types so they can be transferred with `bincode` for `quill`?

Reply:
Yep, didn't know that was a thing :D